keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java

```diff
@@ -281,28 +311,49 @@ private String createSecret(final Tag tag, String secretValue)
     return testSecretName;
   }
 
-  private String createTestSecret(
-      final String tagKey, final String tagVal, final String secretValue) throws Exception {
-    final Tag testSecretTag = Tag.builder().key(tagKey).value(tagVal).build();
+  private boolean checkTestSecretExistsAndUpdateIfUnmatched(
+      final String testSecretName, final Tag tag, final String secretValue) {
     try {
-      return createSecret(testSecretTag, secretValue);
+      final GetSecretValueResponse getSecretValueResponse =
```

Comment:
How is a non-existent secret represented in this response? I suggest we inspect for that and return false before continuing with the code below

Reply:
The request is handled exceptionally so whenever these fail, we know the secret is either non-existent or broken since we're failing to fetch and describe.

Reply:
So there's no way of telling the difference between "exception occurred during request" and "valid request but secret doesn't exist"?

Reply:
Yep, you need to handle that exceptionally with ResourceNotFoundException for example

Reply:
In that case I think we only want to return false in the case of ResourceNotFoundException, all other Exceptions should be thrown